### ISSUE TITLE
tensorflow/core/util/gpu_launch_config.h:129] Check failed: work_element_count > 0 (-1018167296 vs. 0)

### ISSUE THREAD

**Issue by csteel45** [keras-team/tf-keras#124]
I have written a custom Keras CNN-based GAN for synthesizing tabular datasets. The code works fine when I use reasonable batch size (generally 64 to 1024). However, users are allowed to specify a batch size and when they use large ones, I try to handle by catching ResourceExhaustedErrors and step the batch size down. I found that doing this, eventually leads to Check failed error in the post title and I can't catch the exception, the process just dies. This occurs using the following environments:

Windows
Tensorflow 2.6.0 (pip install)
Cuda 11.3
Titan RTX 24GB Founders Edition card
Driver: 465.89

AWS Linux (RHEL7)
Tensorflow 2.6.1 (pip install)
Cuda 11.3 and now 11.5
V100
Driver: 495.29.05

Batch shape is (10240, 328) so at least 2 samples (per related post).
train_step function:
    @tf.function()
    def _train_step(self, real_data):
        noise = tf.random.normal([self._batch_size, self._noise_dim])

        with tf.GradientTape() as gen_tape, tf.GradientTape() as disc_tape:
            synthetic_data = self._generator(noise, training=True)
            real_data_pred = self._discriminator(real_data, training=True)
            synth_data_pred = self._discriminator(synthetic_data, training=True)
            gen_loss = self.generator_loss(synth_data_pred)
            disc_loss = self.discriminator_loss(real_data_pred, synth_data_pred)

        gradients_of_generator = gen_tape.gradient(gen_loss, self._generator.trainable_variables)
        gradients_of_discriminator = disc_tape.gradient(disc_loss, self._discriminator.trainable_variables)
        self.generator_optimizer.apply_gradients(zip(gradients_of_generator, self._generator.trainable_variables))
        self.discriminator_optimizer.apply_gradients(zip(gradients_of_discriminator,
                                                         self._discriminator.trainable_variables))

Also, happens with and without mixed precision.
Any help would be greatly appreciated.

**Comment by bit-pku-zdf** [comment]
I also met this problems.  I use tf1.15.5, and batch is large, can tf development engineers help fix it?

**Comment by tdgstoneman** [comment]
Still having this issue in 2025 with A100 80GB and GH200 96GB. When it happens, GPU ram utilization isn't even at 75%. Tried upgrading from TF 2.15.0 to 2.18.0 and with no luck.
